r3bdata/ttt10Data/R3BTTTXHitData.h

```diff
@@ -0,0 +1,72 @@
+/******************************************************************************
+ *   Copyright (C) 2019 GSI Helmholtzzentrum für Schwerionenforschung GmbH    *
+ *   Copyright (C) 2019-2024 Members of R3B Collaboration                     *
+ *                                                                            *
+ *             This software is distributed under the terms of the            *
+ *                 GNU General Public Licence (GPL) version 3,                *
+ *                    copied verbatim in the file "LICENSE".                  *
+ *                                                                            *
+ * In applying this license GSI does not waive the privileges and immunities  *
+ * granted to it by virtue of its status as an Intergovernmental Organization *
+ * or submit itself to any jurisdiction.                                      *
+ ******************************************************************************/
+
+// -------------------------------------------------------------------
+// -----               R3BTTTXHitData header file                -----
+// -----      Created 11/01/2024 by J. L. Rodriguez-Sanchez      -----
+// -------------------------------------------------------------------
+
+#pragma once
+
+#include <TObject.h>
+#include <cmath>
+#include <cstdint>
+#include <iostream>
+#include <string>
+
+class R3BTTTXHitData : public TObject
+{
+  public:
+    // Default constructor
+    R3BTTTXHitData() = default;
+
+    // Constructor with arguments (explicit)
+    explicit R3BTTTXHitData(uint8_t detID, double xpos, double energy, double ang, double charge, double time = 0);
+
+    // Destructor virtual
+    virtual ~R3BTTTXHitData() = default;
+
+    // Accessors with [[nodiscard]]
+    [[nodiscard]] inline const uint8_t& GetDetID() const { return fDetID; }
+    [[nodiscard]] inline const double& GetX() const { return fXpos; }
+    [[nodiscard]] inline const double& GetEnergy() const { return fEnergy; }
+    [[nodiscard]] inline const double& GetAngle() const { return fAng; }
+    [[nodiscard]] inline const double& GetChargeZ() const { return fCharge; }
+    [[nodiscard]] inline const double& GetTime() const { return fTime; }
+
+    // Modifiers
+    inline void SetDetID(uint8_t id) { fDetID = id; }
+    inline void SetXpos(double xpos) { fXpos = xpos; }
+    inline void SetEnergy(double energy) { fEnergy = energy; }
+    inline void SetAngle(double angle) { fAng = angle; }
+    inline void SetChargeZ(double charge) { fCharge = charge; }
+    inline void SetTime(double time) { fTime = time; }
+
+    // Support for printing
+    [[nodiscard]] std::string toString() const;
+    void Print(const Option_t*) const override;
+
+  protected:
+    uint8_t fDetID = 0;
+    double fXpos = std::nan("");
+    double fEnergy = 0;
+    double fAng = std::nan("");
+    double fCharge = 0;
+    double fTime = 0;
```

Comment:
Why is the reason the default values not being nan for all? It will be anyway defined by cal2hit task though.

Reply:
Because charge = 0, energy=0 and time=0 mean that there is not a particle. Angle =0 or Xpos=0 could be even with ions passing through the detector

Reply:
Everything is nan, this will remove all the zeros from your histograms later.